t/cs.t

```diff
@@ -2056,3 +2089,234 @@ GET /c
 failed to receive: failed to receive the first 2 bytes: closed
 --- error_log
 failed to send close: bad status code
+
+
+=== TEST 29: mutual TLS without client certs
+--- no_check_leak
+--- http_config eval: $::HttpConfig
+--- config
+    listen 12345 ssl;
+    server_name example.com;
+    ssl_certificate ../../cert/mtls_server.crt;
+    ssl_certificate_key ../../cert/mtls_server.key;
+    ssl_client_certificate ../../cert/mtls_ca.crt;
+    ssl_verify_client      on;
+    server_tokens off;
+
+    resolver 127.0.0.1:1953 ipv6=off;
+    resolver_timeout 1s;
+
+    lua_ssl_trusted_certificate ../../cert/mtls_ca.crt;
+    lua_ssl_verify_depth 2;
+
+    location = /c {
+        content_by_lua_block {
+            local client = require "resty.websocket.client"
+            local wb, err = client:new()
+
+            local uri = "wss://example.com:12345/s"
+            local ok, err = wb:connect(uri, { ssl_verify = true })
+            if not ok then
+                ngx.say("failed to connect: " .. err)
+                return
+            end
+
+            local data = "hello"
+            local bytes, err = wb:send_text(data)
+            if not bytes then
+                ngx.say("failed to send frame: ", err)
+                return
+            end
+
+            local typ
+            data, typ, err = wb:recv_frame()
+            if not data then
+                ngx.say("failed to receive 2nd frame: ", err)
+                return
+            end
+        }
+    }
+
+    location = /s {
+        content_by_lua '
+            local server = require "resty.websocket.server"
+            local wb, err = server:new()
+            if not wb then
+                ngx.log(ngx.ERR, "failed to new websocket: ", err)
+                return ngx.exit(444)
+            end
+
+            local data, typ, err = wb:recv_frame()
+            if not data then
+                -- ngx.log(ngx.ERR, "failed to receive a frame: ", err)
+                return ngx.exit(444)
+            end
+
+            -- send it back!
+            local bytes, err = wb:send_text(data)
+            if not bytes then
+                ngx.log(ngx.ERR, "failed to send the 2nd text: ", err)
+                return ngx.exit(444)
+            end
+        ';
+    }
+--- user_files eval: $::mtls_user_files
+--- udp_listen: 1953
+--- udp_reply eval
```

Comment:
Can this be saved as a variable so they the test below can just refer to the function? Like `--- http_config eval: $::HttpConfig`